### ISSUE TITLE
release: 6.0.0 tracking issue

### ISSUE THREAD

**Issue by bastimeyer** [streamlink/streamlink#5401]
**Tracking issue of breaking changes of release 6.0.0**

Official support for py37 ends today, which means we're going to drop support for this old Python version in Streamlink as well.
After #5302 will have been merged into master, this will put the branch into a breaking change state, so other breaking changes will follow.

This doesn't have to be done today though. Streamlink doesn't use the git-flow development model or something similar of that kind, so having breaking changes on the only development+release branch is always risky when other changes aren't ready yet and when we may have to publish quick bugfix releases prior to the intended major version release. We don't want to branch off for patch releases and also want to avoid having unnecessary/ugly revert commits.

There's also the webbrowser implementation which isn't finished yet (I'm getting close though), and I'd like to have it included in 6.0.0, at least in an experimental state, so as long as it's not yet ready, merging breaking changes is probably not a good idea.

----

- [x] #5302 
  Ready
- [x] #5310
  Should be pretty much ready, but I think it could use some documentation (in addition to the announcements in the upcoming changelog)
- [x] #5033 
  Needs a rebase, linting fixes and merge conflict resolving.
- [x] #5402 
- [x] #5403 
- [x] #5404
- [x] #5429
- [x] #5449 

----

Anything else that is currently considered deprecated and could be removed? We don't have deadlines for anything on the deprecations list...
https://streamlink.github.io/deprecations.html

- `Plugin.can_handle_url()` / `Plugin.priority()` looks like a good candidate, especially with the other planned `Plugin` changes.
- `plugin.api.validate.text` as well
- Deprecated CLI args and config paths are rather irrelevant and would only bloat up the changelog for end users who won't read it anyway and confuse them.

**Comment by bastimeyer** [comment]
I will probably also replace the currently used sdist signing key. I don't have the passphrase and in a short conversation with beardypig a couple of months ago, he said that he didn't have it anymore either, which means nobody except GitHub CI via its repo secrets has access to it. Running a special action that sends it to someone else is not a great idea, but could work. Creating a new one is much better though.

**Comment by gravyboat** [comment]
> Creating a new one is much better though.

Agreed, there's no reason to risk it when a new one will get the job done. Let's make sure it is mentioned in the changelog when appropriate.

**Comment by bastimeyer** [comment]
With the basic webbrowser API stuff being finished and submitted in PRs now, we can now go ahead and merge the breaking changes into master.

I will have a look at this later and update the stuff I talked about in the OP.

**Comment by bastimeyer** [comment]
6.0.0 should be ready now except for the signing key update. I still need to figure out how to properly manage this, because I want to avoid a situation like this again where one person sets this up and then forgets/misplaces the passphrase/decryption-key and doesn't share it with anyone else. This now happened twice, the key on TravisCI and now on GH actions (also with one invalid signing key being committed to master once), and in both cases I didn't receive any data, so I was never in control of any of the signing keys. The signing key also should have an expiration date, but in order to properly manage key expirations, a primary key + subkeys should be used, and I need to read up on this first. GPG is kinda bad usability-wise.

I've also been thinking about finally moving the project metadata from `setup.cfg` to `pyproject.toml` (#4441), but I'm not fully sure yet. A couple of things have changed since I opened that PR last year. I'll think about that tomorrow.

Otherwise, I'm going to prepare the changelog soon.

**Comment by altansnl** [comment]
web browser based token scraping does not work on cloud instances. 
@bastimeyer i am assuming this has to do with IP address looking like a cloud IP, hence `bad_bot` will be `true`. and has nothing to do with current implementation.
